### ISSUE TITLE
False positive in colon rule

### ISSUE THREAD

**Issue by nevillco** [realm/SwiftLint#1277]
I'm setting a value to one of two localized strings with a ternary operator like so:

```
let statesTitle = compactEnabled ? Localized.Titles.additionalStates : Localized.Titles.stateLicenses
```

And receiving the SwiftLint colon violation warning:

`Colon violation: colons should be next to the identifier when specifying a type and next to the key in dictionary literals.`

Of course, neither of those situations apply to my above code.

**Comment by marcelofabri** [comment]
I couldn't reproduce it with this snippet

```
$ echo "let statesTitle = compactEnabled ? Localized.Titles.additionalStates : Localized.Titles.stateLicenses" | swiftlint lint --use-stdin  --no-cache --enable-all-rules

Done linting! Found 0 violations, 0 serious in 1 file.
```

**Comment by jpsim** [comment]
@nevillco hi, I'm just following up on this, how can we reproduce this?

**Comment by marcelofabri** [comment]
Closing due lack of activity.

**Comment by vfn** [comment]
@marcelofabri I'm experiencing this same exact issue.

```
$ swiftlint version
0.21.0
```

The following code triggers the `colon` rule:
```Swift
struct FooBar {

    static let foo = true
    static let bar = false

    func baz(lhs: Int, rhs: Int) -> Bool {
        typealias Type = FooBar
        return lhs > rhs ? Type.foo : Type.bar
    }
}
```

Error: 
<img width="1084" alt="screen shot 2017-08-23 at 4 10 53 pm" src="https://user-images.githubusercontent.com/308592/29601221-b01eee9a-881d-11e7-98aa-9e037d0347c3.png">

**Comment by marcelofabri** [comment]
@vfn Thanks for the sample example. I could reproduce it with 0.21.0 as well.

Here's a reduced version of the snippet that still triggers the warning:

```swift
func baz(lhs: Int, rhs: Int) -> Bool {
    typealias Type = FooBar
    return lhs > rhs ? foo : Type.bar
}
```

**Comment by mkchoi212** [comment]
I just took a look at the bug and found something weird.
SourceKitten parses `Type.foo : Type.bar` respectively in a following manner

- Type
`syntaxtype.identifier` <- ❓should be `typeidentifier`
`syntaxtype:typeidentifier`
- Token string 
`foo` <- ❓should be `Type.foo`
`Type.bar`

But even if SourceKitten is able to correctly parse the two tokens as `typeidentifier`s, the switch case in `ColonRule+Type.swift` considers `(.typeidentifier, .typeidentifier)` as not valid kinds.

Any ideas??
